requirements/requirements_asr.txt

```diff
@@ -2,6 +2,7 @@ braceexpand
 editdistance
 einops
 g2p_en
+IPython
```

Comment:
Please revert this one. IPython is optional.

Reply:
I don't think it is. I recently got this error on main after a clean install:

```
Traceback (most recent call last):
  File "/home/dgalvez/scratch/code/asr/open_asr_leaderboard/nemo_asr/run_eval.py", line 12, in <module>
    from nemo.collections.asr.models import ASRModel
  File "/home/dgalvez/scratch/miniconda3/envs/open_asr_leaderboard/lib/python3.10/site-packages/nemo/collections/asr/__init__.py", line 15, in <module>
    from nemo.collections.asr import data, losses, models, modules
  File "/home/dgalvez/scratch/miniconda3/envs/open_asr_leaderboard/lib/python3.10/site-packages/nemo/collections/asr/models/__init__.py", line 22, in <module>
    from nemo.collections.asr.models.clustering_diarizer import ClusteringDiarizer
  File "/home/dgalvez/scratch/miniconda3/envs/open_asr_leaderboard/lib/python3.10/site-packages/nemo/collections/asr/models/clustering_diarizer.py", line 43, in <module>
    from nemo.collections.asr.parts.utils.vad_utils import (
  File "/home/dgalvez/scratch/miniconda3/envs/open_asr_leaderboard/lib/python3.10/site-packages/nemo/collections/asr/parts/utils/vad_utils.py", line 26, in <module>
    import IPython.display as ipd
ModuleNotFoundError: No module named 'IPython'
```

We are unconditionally importing IPython in vad_utils.py. IPython is not intended as a required dependency, but in practice it currently is.

Reply:
Yeah, that's right.  I did consider leaving out IPython, assuming it was intended to be optional.  However, I wanted to point this out and discuss it, so I left it in.

```
$ python -c 'from nemo.collections.asr.models.clustering_diarizer import ClusteringDiarizer'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/code/nemo/t/lib/python3.10/site-packages/nemo/collections/asr/__init__.py", line 15, in <module>
    from nemo.collections.asr import data, losses, models, modules
  File "/code/nemo/t/lib/python3.10/site-packages/nemo/collections/asr/models/__init__.py", line 22, in <module>
    from nemo.collections.asr.models.clustering_diarizer import ClusteringDiarizer
  File "/code/nemo/t/lib/python3.10/site-packages/nemo/collections/asr/models/clustering_diarizer.py", line 43, in <module>
    from nemo.collections.asr.parts.utils.vad_utils import (
  File "/code/nemo/t/lib/python3.10/site-packages/nemo/collections/asr/parts/utils/vad_utils.py", line 985, in <module>
    ) -> ipd.Audio:
NameError: name 'ipd' is not defined. Did you mean: 'pd'?
```

In `vad_utils.py`, there are two plot-related functions which make use of IPython, as return values and in one case, as a declared return type in the function signature.  [`plot()`](https://github.com/NVIDIA/NeMo/blob/r2.0.0rc1/nemo/collections/asr/parts/utils/vad_utils.py#L973-L1066
) and [`plot_sample_from_rttm()`](https://github.com/NVIDIA/NeMo/blob/r2.0.0rc1/nemo/collections/asr/parts/utils/vad_utils.py#L1441-L1481).

I took a quick look..

Those functions not used in many locations.  `plot_sample_from_rttm()` only once.  `plot()` is a bit harder to track down since its name matches `plotlib.plot()`.

```
$ grep -I -l -r -F plot_sample_from_rttm .
./nemo/collections/asr/parts/utils/vad_utils.py
./examples/asr/speech_classification/README.md
./scripts/speaker_tasks/multispeaker_data_analysis.py
```

I have never created a Python package, nor dealt with organizing conditional imports.

I wondered about moving those two functions to a new file, `vad_utils_visualization.py`  or `..._plotting.py`, and changing the usage sites.  This could remove the import in `vad_utils.py`.

But, users still need to be able to install IPython if they want to run notebooks or scripts, so it must be declared somewhere for pip.  Would a new requirements group solve this?  e.g., `pip install nemo_toolkit[extras]`?

If you offer some guidance, I could make the changes on this PR.  Or, I can drop the commit for now, to merge and deal with it in another PR.

Reply:
@ericharper I took out IPython, so this can go into rc1 if you like.  Let me know about reorganizing the code, and I can take a stab in another PR.